src/Pact/Typechecker.hs

```diff
@@ -688,23 +723,26 @@ toAST TBinding {..} = do
     t' <- mangleType aid <$> traverse toUserType t
     an <- trackNode t' aid
     v' <- toAST v
+    let handleLet = do
+          assocAST aid v'
+          return (Named n an aid,v')
     case _tBindType of
-      BindLet -> do
-        assocAST aid v'
-        return (Named n an aid,v')
+      BindLet -> handleLet
       BindSchema _ -> do
         _fieldName <- asPrimString v'
         return (Named n an aid,v')
+      BindInlinedCallArgs -> handleLet
```

Comment:
See this should never occur coming out of pact code, right? Thus wanting to remove it from Term